test/solarwinds_apm/init_test/init_1_test.rb

```diff
@@ -0,0 +1,17 @@
+# Copyright (c) 2019 SolarWinds, LLC.
+# All rights reserved.
+
+require 'minitest_helper'
+$LOAD_PATH.unshift("#{Dir.pwd}/lib/")
+
+describe 'solarwinds_apm_init_1' do
+  it 'SW_APM_ENABLED_set_to_disabled' do
+    puts "\n\033[1m=== TEST RUN: #{RUBY_VERSION} #{File.basename(__FILE__)} #{Time.now.strftime('%Y-%m-%d %H:%M')} ===\033[0m\n"
+
+    log_output = StringIO.new
+    SolarWindsAPM.logger = Logger.new(log_output)
+    ENV['SW_APM_ENABLED'] = 'false'
+    require './lib/solarwinds_apm'
+    assert_includes log_output.string, 'SW_APM_ENABLED environment variable detected and was set to false. SolarWindsAPM disabled'
```

Comment:
These tests are great! Can we also check that in these cases, `SolarWindsAPM.loaded` is false (would there also be a message about the library being disabled for the invalid/missing service key cases below?), and that custom API calls do not cause an error (i.e. no-op API is in place).

Reply:
Added the case when `SolarWindsAPM.loaded` is false, and test if the noop code is loaded. Error msg will be display
```
'SolarWindsAPM not loaded. SolarWinds APM disabled'
'Please check previous log messages.'
```

Since the service key checker was moved up, so it won't reach to the step that load solarwinds if the key is not valid ( "key not valid" log msg will display).

Reply:
Ah I see after looking at `solarwinds_apm.rb` more... `SolarWindsAPM.loaded` is only defined after requiring `oboe_metal` and specifically means if the core library has been loaded, I was confusing it with a more general and probably currently non-existent concept of whether our custom distro (setting up of OTel SDK, bootstrapping of instrumentors, etc) has been initialized.

Reply:
I can make it more clear by requiring otel component earlier than core library, and log if otel component is loaded or not (most case it will be loaded except some of the gem is not installed properly/successfully)